x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/monitoring/MonitoredSystem.java

```diff
@@ -16,6 +16,7 @@
     KIBANA("kibana"),
     LOGSTASH("logstash"),
     BEATS("beats"),
+    ENTERPRISE_SEARCH("enterprise_search"),
```

Comment:
Can you explain the reason for this change a little bit more? This seems like you're now planning on using the monitoring bulk endpoint rather than the regular monitoring endpoint, is that the case?

Reply:
Hi @dakrone !

Metricbeat uses `.monitoring-*` indices to send information to Stack Monitoring from the monitored solutions (Elasticsearch, Kibana, Logstash, Beats). 

There are index templates in Elasticsearch that create data streams for the monitoring information for each solution:

- x-pack/plugin/core/src/main/resources/monitoring-es-mb.json
- x-pack/plugin/core/src/main/resources/monitoring-kibana-mb.json
- x-pack/plugin/core/src/main/resources/monitoring-logstash-mb.json
- x-pack/plugin/core/src/main/resources/monitoring-beats-mb.json

We want to align Enterprise Search with the rest of the stack solutions that use monitoring, and have monitoring information available from Stack Monitoring. So, we will add `xpack.enabled` to the Enterprise Search Metricbeat module (https://github.com/elastic/beats/pull/29871) and create the necessary data streams using these index templates.

Hope it makes sense!

Reply:
@carlosdelest I think I understand your general use, but I still don't understand whether you're using the regular bulk endpoint, or the `_monitoring/bulk` endpoint. The reason I'm asking is that outside of bugfixes monitoring is not really touched, but this adds a new system to the `_monitoring/bulk` endpoint. I was under the impression that we'd be switching away from that endpoint for things like metricbeat monitoring, and thus this change wouldn't be required.

Reply:
Got it, thanks @dakrone ! I didn't realise that this was related to the `_monitoring/bulk` endpoint; I was looking to have parity with the current monitored solutions.

I have removed the changes from `MonitoredSystem` and `MonitoringIT`. Could you please review it now to ensure nothing is added to deprecated monitoring features?

Thanks!